ffi/android/DemoApp/app/src/main/java/org/lnpbp/demoapp/DemoApp.java

```diff
@@ -0,0 +1,31 @@
+package org.lnpbp.demoapp;
+
+import android.app.Application;
+import android.util.Log;
+
+import org.lnpbp.rgbnode.Runtime;
+
+import java.util.HashMap;
+
+public class DemoApp extends Application {
+    private Runtime runtime;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        Log.i("RGB_NODE", "loading library");
+        System.loadLibrary("rgb_node");
+
+        final String datadir = getFilesDir().toString();
+        final String network = "testnet";
+
+        final HashMap contractEndpoints = new HashMap();
+        contractEndpoints.put("Fungible", String.format("ipc:%s/%s/fungibled.ipc", datadir, network));
+        this.runtime = new Runtime(network, String.format("ipc:%s/%s/stashd.ipc", datadir, network), contractEndpoints, true, datadir);
```

Comment:
```suggestion
        contractEndpoints.put("Fungible", String.format("ipc:%s/%s/fungibled.rpc", datadir, network));
        this.runtime = new Runtime(network, String.format("ipc:%s/%s/stashd.rpc", datadir, network), contractEndpoints, true, datadir);
```

The socket addresses are not passed down to individual threads through config (they are not captured by `daemon` arg conversion process. Thus, it will work only if you use RPC & filenames which are default for thread configs.

This is a quickfix; we need to make the configuration to pass - and than even `inproc` ZMQ will work smoothly